matrix_sdk_base/src/store/sled_store/mod.rs

```diff
@@ -587,6 +587,17 @@ impl SledStore {
             .transpose()?
             .unwrap_or_default())
     }
+
+    pub async fn get_account_data_event(
+        &self,
+        event_type: EventType,
+    ) -> Result<Option<AnyBasicEvent>> {
+        Ok(self
+            .account_data
+            .get(event_type.to_string().as_str().encode())?
+            .map(|m| self.deserialize_event(&m))
+            .transpose()?)
```

Comment:
Can `Ok(<expr>?)` not be simplified to `<expr>` here?

Reply:
No, they are not the same error type:

```
mismatched types
expected enum `std::result::Result<std::option::Option<AnyBasicEvent>, StoreError>`
   found enum `std::result::Result<std::option::Option<_>, SerializationError>`
```

Reply:
Could write `<expr>.map_err(Into::into)`, but probably not worth it.